src/server/scripts/Outland/CoilfangReservoir/TheSlavePens/boss_ahune.cpp

```diff
@@ -0,0 +1,1056 @@
+    {
+        return new spell_slippery_floor_periodic_SpellScript();
+    }
+};
+
+// 46146 - Ahune Spanky Hands
+class spell_ahune_spanky_hands : public SpellScriptLoader
+{
+public:
+    spell_ahune_spanky_hands() : SpellScriptLoader("spell_ahune_spanky_hands") { }
+
+    class spell_ahune_spanky_hands_AuraScript : public AuraScript
+    {
+        PrepareAuraScript(spell_ahune_spanky_hands_AuraScript);
+
+        bool Validate(SpellInfo const* /*spellInfo*/) override
+        {
+            if (!sSpellMgr->GetSpellInfo(SPELL_COLD_SLAP))
+                return false;
+            return true;
+        }
+
+        void HandleProc(AuraEffect const* /*aurEff*/, ProcEventInfo& eventInfo)
+        {
+            PreventDefaultAction();
+            GetTarget()->CastSpell(eventInfo.GetProcTarget(), SPELL_COLD_SLAP, true);
+        }
+
+        void Register() override
+        {
+            OnEffectProc += AuraEffectProcFn(spell_ahune_spanky_hands_AuraScript::HandleProc, EFFECT_0, SPELL_AURA_PROC_TRIGGER_SPELL);
+        }
+    };
+
+    AuraScript* GetAuraScript() const override
+    {
+        return new spell_ahune_spanky_hands_AuraScript();
+    }
+};
+
+class spell_ahune_minion_despawner : public SpellScriptLoader
+{
+public:
+    spell_ahune_minion_despawner() : SpellScriptLoader("spell_ahune_minion_despawner") { }
+
+    class spell_ahune_minion_despawner_SpellScript : public SpellScript
+    {
+        PrepareSpellScript(spell_ahune_minion_despawner_SpellScript);
+
+        void HandleScript(SpellEffIndex /*effIndex*/)
+        {
+            if (GetHitCreature())
+                GetHitCreature()->DespawnOrUnsummon();
+        }
+
+        void Register() override
+        {
+            OnEffectHitTarget += SpellEffectFn(spell_ahune_minion_despawner_SpellScript::HandleScript, EFFECT_0, SPELL_EFFECT_APPLY_AURA);
+        }
+    };
+
+    SpellScript* GetSpellScript() const override
+    {
+        return new spell_ahune_minion_despawner_SpellScript();
+    }
+};
+
+/* TODO
+//The Hook OnEffectHitTarget isn't getting called ever, casting in the target selection hook works, but it's unacceptable
+// 46398 - Spell Ice Bombardment Dest Picker
+class spell_ice_bombardment_dest_picker : public SpellScriptLoader
+{
+public:
+    spell_ice_bombardment_dest_picker() : SpellScriptLoader("spell_ice_bombardment_dest_picker") { }
+
+    class spell_ice_bombardment_dest_picker_SpellScript : public SpellScript
+    {
+        PrepareSpellScript(spell_ice_bombardment_dest_picker_SpellScript);
+
+        bool Validate(SpellInfo const* spellInfo) override
+        {
+            if (!sSpellMgr->GetSpellInfo(SPELL_ICE_BOMBARDMENT))
+                return false;
+            return true;
+        }
+
+        void HandleScriptEffect(SpellEffIndex effIndex)
+        {
+            if (GetCaster())
+                GetCaster()->CastSpell(GetHitUnit(), SPELL_ICE_BOMBARDMENT, true);
+        }
+
+        void Register()
```

Comment:
the spell does not hit a object target, even spell target conditions require one, the spell will target a destination which is the position of the object being selected by conditions

you have to use OnEffectHit hook and cast the spell to GetHitDest() (use Unit::CastSpell overload with x, y, z parameters)